frontend/src/lib/components/NavPanelAdvertisement/NavPanelAdvertisement.tsx

```diff
@@ -0,0 +1,136 @@
+import { useActions, useValues } from 'kea'
+import posthog from 'posthog-js'
+
+import { IconX } from '@posthog/icons'
+import { Link, Tooltip } from '@posthog/lemon-ui'
+
+import { FEATURE_FLAGS } from 'lib/constants'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { featureFlagLogic } from 'lib/logic/featureFlagLogic'
+import { availableOnboardingProducts } from 'scenes/onboarding/utils'
+import { getProductIcon } from 'scenes/products/Products'
+
+import { sidePanelStateLogic } from '~/layout/navigation-3000/sidepanel/sidePanelStateLogic'
+import { SidePanelTab } from '~/types'
+
+import { FlaggedFeature } from '../FlaggedFeature'
+import { navPanelAdvertisementLogic } from './NavPanelAdvertisementLogic'
+
+type Payload = {
+    product_key: string
+    header?: string
+    text?: string
+    docs_link?: string
+    app_link?: string
+    product_info_link?: string
+}
+
+export function NavPanelAdvertisement(): JSX.Element | null {
+    const { featureFlags } = useValues(featureFlagLogic)
+
+    if (featureFlags[FEATURE_FLAGS.TARGETED_PRODUCT_UPSELL] === 'control') {
+        return null
+    }
+
+    return (
+        <FlaggedFeature
+            flag={FEATURE_FLAGS.TARGETED_PRODUCT_UPSELL}
+            children={(_flagValue, payload: Payload) => {
+                if (payload === undefined) {
+                    return null
+                }
+                return <NavPanelAdvertisementContent payload={payload} />
+            }}
+        />
+    )
+}
+
+export function NavPanelAdvertisementContent({ payload }: { payload: Payload }): JSX.Element | null {
+    const product = availableOnboardingProducts[payload.product_key as keyof typeof availableOnboardingProducts]
+    const logic = navPanelAdvertisementLogic({ productKey: payload.product_key })
+    const { hideAdvertisement } = useActions(logic)
+    const { hidden } = useValues(logic)
+
+    if (hidden || (!product && !payload.header)) {
+        return null
+    }
+
+    return (
+        <div className="w-full my-1">
+            <Tooltip
+                title="Based on your usage we think you might like this other product
+                            we have, so we're gently bringing it to your attention :) Feel free to 
+                            dismiss it to make it go away."
+            >
+                <p className="text-xxs text-muted mb-1 p-0 text-right">ads via PostHog</p>
```

Comment:
Could probably do with a small info icon to make it clearer there's a tooltip. I wouldn't think of hovering this tbh.